protocols/872-cosmosid-ngs-library-prep-part1/nexteraXT_dna_library_prep_part1.ot2.py

```diff
@@ -0,0 +1,104 @@
+from opentrons import labware, instruments, robot
+
+"""
+Tagment Genomic DNA
+Amplify Libraries
+"""
+# labware setup
+gDNA_plate = labware.load('96-PCR-flat', '1')
+out_plate = labware.load('96-PCR-flat', '2')
+# out_plate = labware.load('biorad-hardshell-96-PCR', '2')
+tuberack = labware.load('tube-rack-2ml', '5')
+
+# reagent setup
+atm = tuberack.wells('A1')  # Amplicon Tagment Mix
+td = tuberack.wells('B1')  # Tagment DNA Buffer
+nt = tuberack.wells('C1')  # Neutralize Tagment Buffer
+npm = tuberack.wells('D1')  # Nextera PCR Master Mix
+index_1 = tuberack.wells('A2', to='D4')  # Index 1 (i7) adapters
+index_2 = tuberack.wells('A5', to='D6')  # Index 2 (i5) adapters
+
+tipracks50 = [labware.load('tiprack-200ul', slot) for slot in ['3', '4']]
+tipracks10 = [labware.load('tiprack-10ul', slot) for slot in ['6', '7']]
+
+# pipette setup
+p50 = instruments.P50_Single(
+    mount='left',
+    tip_racks=tipracks50)
+
+p10 = instruments.P10_Single(
+    mount='right',
+    tip_racks=tipracks10)
+
+
+def run_custom_protocol(
+        number_of_samples: int=24):
+
+    # define sample locations
+    samples = gDNA_plate.wells()[:number_of_samples]
+
+    index1 = {name: [] for name in range(len(index_1))}
+    index2 = {name: [] for name in range(len(index_2))}
+
+    if number_of_samples <= 24:
```

Comment:
Instead of doing this, you will determine where in the plate you are pipetting by the _length_ of the adapters (found based on number of samples).

So...if you have 5 i7 adapters and 4 i5 adapters then you already know based on the length that you are dispensing into **5** columns and **4** rows.